File: Arrays  101/283. Move Zeros/move_zeros.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        read_ptr, write_ptr = 0, 0
        while read_ptr < len(nums):
            if nums[read_ptr] == 0:
                read_ptr += 1
                continue
            nums[write_ptr] = nums[read_ptr]
            write_ptr += 1
            read_ptr += 1

        while write_ptr < len(nums):
            nums[write_ptr] = 0
            write_ptr += 1

File: Arrays  101/283. Move Zeros/move_zeros.cpp
class Solution
{
public:
    void moveZeroes(vector<int> &nums)
    {
        int read_ptr = 0;
        int write_ptr = 0;

        while (read_ptr < nums.size())
        {
            if (nums[read_ptr] == 0)
            {
                read_ptr++;
                continue;
            }
            nums[write_ptr++] = nums[read_ptr++];
        }
        while (write_ptr < nums.size())
        {
            nums[write_ptr++] = 0;
        }
    }
};